src/service/VotingService.ts

```diff
@@ -14,108 +14,107 @@
-
-            const cmd = [
-                `${presetData.catapultAppFolder}/bin/catapult.tools.votingkey`,
-                `--secret=${votingPrivateKey}`,
-                `--startEpoch=${presetData.votingKeyStartEpoch}`,
-                `--endEpoch=${presetData.votingKeyEndEpoch}`,
-                `--output=/votingKeys/${privateKeyTreeFileName}`,
-            ];
-            await BootstrapUtils.deleteFolder(votingKeysFolder);
             await BootstrapUtils.mkdir(votingKeysFolder);
-            const binds = [`${votingKeysFolder}:/votingKeys:rw`];
+            const votingUtils = new VotingUtils();
+            await BootstrapUtils.deleteFile(join(votingKeysFolder, 'metadata.yml'));
+            let newFileCreated = false;
+            while (true) {
+                const currentVotingFiles = votingUtils.loadVotingFiles(votingKeysFolder);
+                const maxVotingKeyEndEpoch = currentVotingFiles[currentVotingFiles.length - 1]?.endEpoch || networkEpoch - 1;
+                const votingKeyDesiredFutureLifetime = presetData.votingKeyDesiredFutureLifetime || presetData.votingKeyDesiredLifetime / 2;
+                nodeAccount.voting = currentVotingFiles;
+                if (maxVotingKeyEndEpoch > networkEpoch + votingKeyDesiredFutureLifetime) {
+                    return newFileCreated;
+                }
+                if (!update && currentVotingFiles.length > 0) {
+                    logger.warn(
+                        `Voting key files are close to EXPIRATION or have EXPIRED!. Run the 'symbol-bootstrap upgradeVotingKeys' command!`,
+                    );
+                    logger.warn('');
+                    return newFileCreated;
+                }
+                const votingKeyStartEpoch = maxVotingKeyEndEpoch + 1;
+                const votingKeyEndEpoch = maxVotingKeyEndEpoch + presetData.votingKeyDesiredLifetime;
+                const votingAccount = Account.generateNewAccount(presetData.networkType);
+                const votingPrivateKey = votingAccount.privateKey;
+                const epochs = votingKeyEndEpoch - votingKeyStartEpoch + 1;
+                logger.info(`Creating Voting key file of ${epochs} epochs for node ${nodeAccount.name}. This could take a while!`);
+                const privateKeyTreeFileName = `private_key_tree${currentVotingFiles.length + 1}.dat`;
+                if (presetData.useExperimentalNativeVotingKeyGeneration) {
+                    logger.info('Voting file is created using the native typescript voting key file generator!');
+                    const votingFile = await votingUtils.createVotingFile(votingPrivateKey, votingKeyStartEpoch, votingKeyEndEpoch);
+                    writeFileSync(join(votingKeysFolder, privateKeyTreeFileName), votingFile);
+                } else {
+                    logger.info(`Voting file is created using docker and the default's catapult.tools.votingkey`);
+                    const binds = [`${votingKeysFolder}:/votingKeys:rw`];
```

Comment:
Should the votingService gets called here?

Reply:
I don't understand, could you rephrase?